### PR TITLE
Change SurfaceCapabilities fields to use RangeInclusive instead of Range

### PR DESCRIPTION
**Pull request by rukai**
Fixes https://github.com/gfx-rs/gfx/issues/2571
PR checklist:
- [x] `make` succeeds (on *nix)
- [x] `make reftests` succeeds
- [x] tested examples with the following backends:
vulkan - linux
gl - linux

~~`rustfmt` run on changed code~~

### CODE REVIEW

File: src/backend/gl/src/window/glutin.rs [hunk 1]
```diff
@@ -135,15 +135,15 @@ impl hal::Surface<B> for Surface {
     ) {
         let caps = hal::SurfaceCapabilities {
             image_count: if self.context.get_pixel_format().double_buffer {
-                2 .. 3
+                2 ..= 3
```
Comment (kvark) [inline src/backend/gl/src/window/glutin.rs]: this should be `2 ..= 2` and the other one - `1 ..= 1`

File: src/backend/metal/src/window.rs [hunk 2]
```diff
@@ -326,11 +326,11 @@ impl hal::Surface<Backend> for Surface {
             device_caps.os_is_mac || device_caps.has_version_at_least(11, 2);
 
         let image_count = if can_set_maximum_drawables_count {
-            2 .. 4
+            2 ..= 4
```
Comment (kvark) [inline src/backend/metal/src/window.rs]: this should be `2 ..= 3` and the other one is `3 ..= 3`

File: src/hal/src/window.rs [hunk 3]
```diff
@@ -311,9 +311,9 @@ impl SwapchainConfig {
         let clamped_extent = match caps.current_extent {
             Some(current) => current,
             None => {
-                let (min_width, max_width) = (caps.extents.start.width, caps.extents.end.width - 1);
+                let (min_width, max_width) = (caps.extents.start().width, caps.extents.end().width - 1);
```
Comment (kvark) [inline src/hal/src/window.rs]: we need to remove "-1" from all these now

File: src/backend/dx12/src/window.rs [hunk 4]
```diff
@@ -74,9 +74,9 @@ impl hal::Surface<Backend> for Surface {
         let extent = hal::window::Extent2D { width, height };
 
         let capabilities = hal::SurfaceCapabilities {
-            image_count: 2 .. 16, // we currently use a flip effect which supports 2..16 buffers
+            image_count: 2 ..= 16, // we currently use a flip effect which supports 2..=16 buffers
```
Comment (rukai) [inline src/backend/dx12/src/window.rs]: Just want to double check this is correct? No idea what a "flip effect" is.

**Comment by rukai** [comment]
Oh wow, I totally missed some of the backends where providing exclusive ranges.
Everything should be addressed now.

**Comment by kvark** [comment]
Bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by rukai** [comment]
Can we retry this again?

**Comment by kvark** [comment]
Bors retry

**Comment by rukai** [comment]
Whats up with bors? I dont have access to the details page.

**Comment by kvark** [comment]
Bors +
Bors retry

**Pull request merged**